dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -1261,9 +1275,11 @@ public void saveActionClass(WorkflowActionClass actionClass) throws DotDataExcep
 		// cache.remove(step);
 
 		// update workflowScheme mod date
-		WorkflowAction action = findAction(actionClass.getActionId());
-		WorkflowScheme scheme = findScheme(action.getSchemeId());
+		final WorkflowAction action = findAction(actionClass.getActionId());
+		final WorkflowScheme scheme = findScheme(action.getSchemeId());
+
 		saveScheme(scheme);
+		cache.remove(action);

```

Comment:
We should invalidate the Actionlets also when an Action is removed and maybe when a Scheme is removed.